src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ADD_CAPACITY = 1.5;
+    private static final int POSITION_ZERO = 0;
+    private T [] elements;
+    private int listSize;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void checkIndexForExistingElements(int index) {
```

Comment:
```suggestion
    private void checkIndex(int index) {
```

private methods should be placed after all public methods